Makefile

```diff
@@ -1,70 +1,222 @@
+BUNDLE_IMG ?= $(IMAGE_TAG_BASE)-bundle:v$(TOPOLVM_OPERATOR_VERSION)
+
+IMG ?= $(REGISTRY_PREFIX)/$(BUNDLE_IMG)
+
+# ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
+# ENVTEST_K8S_VERSION = 1.22
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+# Setting SHELL to bash allows bash commands to be executed by recipes.
+# This is a requirement for 'setup-envtest.sh' in the test target.
+# Options are set to exit when a recipe line exits non-zero or a piped command fails.
+SHELL = /usr/bin/env bash -o pipefail
+.SHELLFLAGS = -ec
+
+all: build
+
+##@ General
+
+# The help target prints out all targets with their descriptions organized
+# beneath their categories. The categories are represented by '##@' and the
+# target descriptions by '##'. The awk commands is responsible for reading the
+# entire set of makefiles included in this invocation, looking for lines of the
+# file as xyz: ## something, and then pretty-format the target and help. Then,
+# if there's a line with ##@ something, that gets pretty-printed as a category.
+# More info on the usage of ANSI control characters for terminal formatting:
+# https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_parameters
+# More info on the awk command:
+# http://linuxcommand.org/lc3_adv_awk.php
+
+help: ## Display this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+##@ Development
+
+manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
 	$(CONTROLLER_GEN) \
 		crd:crdVersions=v1 \
 		rbac:roleName=topolvm-global \
 		paths="./api/...;./controllers" \
 		output:crd:artifacts:config=config/crd/bases
 
-image:
-	docker build -t $(IMAGE_PREFIX)topolvm-operator:devel --build-arg TOPOLVM_OPERATOR_VERSION=$(TOPOLVM_OPERATOR_VERSION) .
-
-tag:
-	docker tag $(IMAGE_PREFIX)topolvm-operator:devel $(IMAGE_PREFIX)topolvm-operator:$(IMAGE_TAG)
-
-push:
-	docker push $(IMAGE_PREFIX)topolvm-operator:$(IMAGE_TAG)
+generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./api/..."
 
-# Run go fmt against code
-fmt:
+fmt: ## Run go fmt against code.
 	go fmt ./...
 
-# Run go vet against code
-vet:
+vet: ## Run go vet against code.
 	go vet ./...
 
-# Generate code
-generate:
-	$(CONTROLLER_GEN) object:headerFile=./hack/boilerplate.go.txt paths="./api/..."
+test: manifests generate fmt vet envtest ## Run tests.
+	@KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go test ./... -coverprofile cover.out
```

Comment:
this line should be deleted,  make test should not run env test.

and run ut  must use privileged mode

Reply:
- `envtest` isn't actually running here
- I just commented out first line, yet to reach consense on what targets does the Makefile should have.

Reply:
> 
OK， but it will trigger ut.